### PR TITLE
make use of istio hash pkg

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

It is more efficient than golang hash as depicted by #42715

### CODE REVIEW

File: pilot/pkg/serviceregistry/serviceentry/controller.go [hunk 1]
```diff
@@ -904,10 +904,11 @@ func autoAllocateIPs(services []*model.Service) []*model.Service {
 		// 3. the hostname is not a wildcard
 		if svc.DefaultAddress == constants.UnspecifiedIP && !svc.Hostname.IsWildCarded() &&
 			svc.Resolution != model.Passthrough {
-			hash.Write([]byte(makeServiceKey(svc)))
+			h.Reset()
+			h.Write([]byte(makeServiceKey(svc)))
 			// First hash is calculated by
-			s := hash.Sum32()
-			firstHash := s % uint32(maxIPs)
+			s := h.Sum64()
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/serviceentry/controller.go:910]: I can not remember, but I ran in to some issue with Sum64 because of size.

**Review by howardjohn — COMMENTED** [review]
overall lgtm, not sure about Rama's comment though so will leave it to them to approve

File: pilot/pkg/serviceregistry/serviceentry/controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: "github.com/cespare/xxhash/v2" does not support uint32 at all.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: It is better if you can recall what's the problem. IMO. uint64 and uint32 has no much difference in this case, we will mod maxIps(255 * 255) here. the conflict probability is similar in theory.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: I know what the problem, it seems we can never cal all values covering [0, 255*255)

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: will look into how we can change the secondary hash

Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: correct. That is why I used 32.

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/controller.go:910]: Tested manually, sum64 is slower than sum32.

**Comment by hzxuzhonghu** [comment]
And also i find the collision number is very large. for `Test_autoAllocateIP_values`, the conflict number is `2754901`

And the case running takes 0.18s in my local VM. I think this is unacceptable. Also I tried many other tests cases:

```
255*255 services, it takes 0.18s
125*255(31875)               0.081s
```

I think we need some benchmark for it

**Pull request closed without merging**